src/cork/client.rs

```diff
@@ -1,31 +1,67 @@
+use std::collections::{HashMap, HashSet};
+
 use crate::{error::Error, prelude::APP};
 use abscissa_core::Application;
-use somm_proto::cork::{
-    query_client::QueryClient, QueryCellarIDsRequest, QueryCellarIDsResponse,
-    QueryScheduledCorksByIdRequest, QueryScheduledCorksByIdResponse,
+use somm_proto::{
+    axelar_cork::{
+        query_client::QueryClient as AxelarQueryClient,
+        QueryCellarIDsRequest as AxelarQueryCellarIDsRequest,
+        QueryScheduledCorksByIdRequest as AxelarQueryScheduledCorksByIdRequest,
+        QueryScheduledCorksByIdResponse as AxelarQueryScheduledCorksByIdResponse,
+    },
+    cork::{
+        query_client::QueryClient, QueryCellarIDsRequest, QueryScheduledCorksByIdRequest,
+        QueryScheduledCorksByIdResponse,
+    },
 };
 use tonic::{transport::Channel, Response, Status};
 
 /// Client wrapper used to refresh the cache
 pub struct CorkQueryClient {
-    inner: QueryClient<Channel>,
+    inner_cork: QueryClient<Channel>,
+    inner_axelar: AxelarQueryClient<Channel>,
 }
 
 impl CorkQueryClient {
     /// Constructor
     pub async fn new() -> Result<Self, Error> {
         let config = APP.config();
         Ok(CorkQueryClient {
-            inner: QueryClient::connect(config.cosmos.grpc.clone()).await?,
+            inner_cork: QueryClient::connect(config.cosmos.grpc.clone()).await?,
+            inner_axelar: AxelarQueryClient::connect(config.cosmos.grpc.clone()).await?,
         })
     }
 
     /// Queries the Sommelier chain for approved Cellar IDs
     pub async fn get_approved_cellar_ids(
         &mut self,
-    ) -> Result<Response<QueryCellarIDsResponse>, Status> {
-        let request = QueryCellarIDsRequest {};
-        self.inner.query_cellar_i_ds(request).await
+    ) -> Result<HashMap<u64, HashSet<String>>, Status> {
+        let cork_request = QueryCellarIDsRequest {};
+        let cork_result = self.inner_cork.query_cellar_i_ds(cork_request).await?;
+        let axelarcork_request = AxelarQueryCellarIDsRequest {};
+        let axelarcork_result = self
+            .inner_axelar
+            .query_cellar_i_ds(axelarcork_request)
+            .await?;
+
+        let mut result: HashMap<u64, HashSet<String>> = HashMap::new();
+        for id in cork_result.into_inner().cellar_ids {
+            let normalized_id = id.trim().to_lowercase();
```

Comment:
Same thing here: we don't want to be adjusting the strings we get back from the chain. When they are written to the chain they are converted to Eth addresses that output in the checksum format.